flocker/dockerplugin/test/test_api.py

```diff
@@ -0,0 +1,124 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for the Volumes Plugin API provided by the plugin.
+"""
+
+from uuid import uuid4, UUID
+
+from twisted.web.http import OK
+
+from .._api import VolumePlugin, DEFAULT_SIZE
+from ...apiclient import FakeFlockerClient, Dataset
+from ...control._config import dataset_id_from_name
+
+from ...restapi.testtools import buildIntegrationTests, APIAssertionsMixin
+
+
+class APITestsMixin(APIAssertionsMixin):
+    """
+    Helpers for writing tests for the Docker Volume Plugin API.
+    """
+    NODE_A = uuid4()
+    NODE_B = uuid4()
+
+    def initialize(self):
+        """
+        Create initial objects for the ``VolumePlugin``.
+        """
+        self.flocker_client = FakeFlockerClient()
+
+    def test_pluginactivate(self):
+        """
+        ``/Plugins.Activate`` indicates the plugin is a volume driver.
+        """
+        # Really we should be sending a blank body, but that has some
+        # issues since @structured then expects a POST to have a
+        # application/json content type. Fixing up the content type issues
+        # (a necessary chunk of work) is covered by FLOC-2811, which
+        # should also fix this.
+        return self.assertResult(b"POST", b"/Plugin.Activate", {}, OK,
+                                 {u"Implements": [u"VolumeDriver"]})
+
+    def test_remove(self):
+        """
+        ``/VolumeDriver.Remove`` returns a successful result.
+        """
+        return self.assertResult(b"POST", b"/VolumeDriver.Remove",
+                                 {u"Name": u"vol"}, OK, {u"Err": None})
+
+    def test_unmount(self):
+        """
+        ``/VolumeDriver.Unmount`` returns a successful result.
+        """
+        return self.assertResult(b"POST", b"/VolumeDriver.Unmount",
+                                 {u"Name": u"vol"}, OK, {u"Err": None})
+
+    def test_create_creates(self):
+        """
+        ``/VolumeDriver.Create`` creates a new dataset in the configuration.
+        """
+        name = u"myvol"
+        d = self.assertResult(b"POST", b"/VolumeDriver.Create",
+                              {u"Name": name}, OK, {u"Err": None})
```

Comment:
:question: I'd normally have a separate test that asserts that the initial `list_datasets_configuration` was empty. Not strictly necessary here, but add it if you feel that it helps.